src/analyze/analyze-verify.c

```diff
@@ -233,6 +248,16 @@ int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run
         if (strv_isempty(filenames))
                 return 0;
 
+        /* Allow systemd-analyze to hook in a callback function so that it can get
+         * all the required log data from the function itself without having to rely
+         * on a global set variable for the same */
+        if (return_error_on != RETURN_ERROR_ON_NONE) {
+                s = set_new(&string_hash_ops);
+                if (!s)
+                        return -ENOMEM;
+                set_log_syntax_callback(&log_syntax_callback, &s);
+        }
```

Comment:
please undo the setting of the callback function at the end (maybe via _cleanup_ on a throw-away local variable, so that it is done automatically when things get out of scope)